backend/api/api.go

```diff
@@ -138,20 +148,26 @@ func DeleteTodoItem(c *gin.Context) {
 		c.JSON(http.StatusNotAcceptable, gin.H{"message": "please enter an id"})
 	} else {
 		// Find and delete the todo item
-		_, err := db.Query("DELETE FROM list WHERE id=$1;", id)
-		if err != nil {
+		var exists bool
+		err := db.QueryRow("SELECT * FROM list WHERE id=$1;", id).Scan(&exists)
+		if err != nil && err == sql.ErrNoRows {
 			fmt.Println(err.Error())
-			c.JSON(http.StatusInternalServerError, gin.H{"message": "error with DB"})
-			panic(err)
+			c.JSON(http.StatusNotFound, gin.H{"message": "not found"})
+		} else {
+			_, err = db.Query("DELETE FROM list WHERE id=$1;", id)
+			if err != nil {
+				fmt.Println(err.Error())
+				c.JSON(http.StatusInternalServerError, gin.H{"message": "error with DB"})
+			}
+
+			// Log message
+			log.Println("deleted todo item", id)
+
+			// Return success response
+			c.Header("Access-Control-Allow-Origin", "*")
+			c.Header("Access-Control-Allow-Headers", "access-control-allow-origin, access-control-allow-headers")
+			c.JSON(http.StatusOK, gin.H{"message": "successfully deleted todo item", "todo": id})
```

Comment:
You could use 204 status code here and for put and 201 for successful create.

Reply:
Oh yes, I did not know about 204 and 201. I will update the successful status codes to them now.